src/inspector_profiler.cc

```diff
@@ -76,34 +81,74 @@ void V8ProfilerConnection::V8ProfilerSessionDelegate::SendMessageToFrontend(
     const v8_inspector::StringView& message) {
   Environment* env = connection_->env();
   Isolate* isolate = env->isolate();
+  Local<Context> context = env->context();
   HandleScope handle_scope(isolate);
-  Context::Scope context_scope(env->context());
+  Context::Scope context_scope(context);
 
-  // TODO(joyeecheung): always parse the message so that we can use the id to
-  // identify ending messages as well as printing the message in the debug
-  // output when there is an error.
   const char* type = connection_->type();
-  Debug(env,
-        DebugCategory::INSPECTOR_PROFILER,
-        "Receive %s profile message, ending = %s\n",
-        type,
-        connection_->ending() ? "true" : "false");
-  if (!connection_->ending()) {
-    return;
-  }
-
   // Convert StringView to a Local<String>.
   Local<String> message_str;
   if (!String::NewFromTwoByte(isolate,
                               message.characters16(),
                               NewStringType::kNormal,
                               message.length())
            .ToLocal(&message_str)) {
-    fprintf(stderr, "Failed to convert %s profile message\n", type);
+    fprintf(
```

Comment:
it would be nice if this was available as an error callback or something, since the calling code can't really read stderr

Reply:
The only reason I could see it failing is when `message.length() > kStringMaxLength` which:

1. I don't think can realistically happen, and
2. Isn't actionable for users when it does (except by opening a bug report here)

Maybe the error message could include the size and perhaps the first `N` bytes of the message.

Similar reasoning applies to the error code paths below. Failure likely means we need to fix something on our end.